File: JsoupWebCrawling/app/src/main/java/com/example/jsoupwebcrawling/MainActivity.java
package com.example.jsoupwebcrawling;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;
import androidx.core.content.PermissionChecker;

import android.Manifest;
import android.content.pm.PackageManager;
import android.os.AsyncTask;
import android.os.Bundle;
import android.text.method.ScrollingMovementMethod;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    private String TargetURL = "https://sports.news.naver.com/index.nhn";
    private TextView tv1;
    private String htmlContentInStringFormat="";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv1 = (TextView)findViewById(R.id.tv1);
        tv1.setMovementMethod(new ScrollingMovementMethod()); //스크롤 가능한 텍스트뷰로 만들기

        Button btn1 = (Button)findViewById(R.id.btn1);
        btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                tv1.setText("");
                htmlContentInStringFormat = "";

                JsoupAsyncTask jsoupAsyncTask = new JsoupAsyncTask();
                jsoupAsyncTask.execute();
            }
        });
    }

    private class JsoupAsyncTask extends AsyncTask<Void, Void, Void> {

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected Void doInBackground(Void... params) {
            try {
                Document doc = Jsoup.connect(TargetURL).get();

                //테스트1
                Elements titles= doc.select("li.today_item strong.title");

                System.out.println("-------------------------------------------------------------");
                for(Element e: titles){
                    System.out.println("title: " + e.text());
                    htmlContentInStringFormat += e.text().trim() + "\n";
                }
            } catch (IOException e) {
                Toast.makeText(getApplicationContext(), "정보를 불러오는데 실패했습니다.", Toast.LENGTH_SHORT).show();
                e.printStackTrace();
            }

            return null;
        }

        @Override
        protected void onPostExecute(Void result) {
            tv1.setText(htmlContentInStringFormat);
        }

    }
}

File: RoomDB/app/src/main/java/com/example/roomdb/MainActivity.kt
package com.example.roomdb

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.MutableLiveData
import com.example.roomdb.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    val email = MutableLiveData<String>()
    val password = MutableLiveData<String>()

    lateinit var binding : ActivityMainBinding

    lateinit var dao : Dao
    private lateinit var  userList: List<UserDataBase>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.main = this

        dao = DataBase.getInstance(this)!!.dao()

        setText()
    }

    fun setText() {
        userList = dao.getAll()

        if (userList.isNotEmpty()) {
            email.value = userList[0].email
            password.value = userList[0].password
        }
    }

    fun login() {
        dao.delete()
        dao.insert(UserDataBase(
            id = 0,
            email = email.value.toString(),
            password = password.value.toString()
            )
        )

        Toast.makeText(this, "성공", Toast.LENGTH_SHORT).show()
    }
}